content/docs/api-reference/make_checkpoint.md

```diff
@@ -0,0 +1,83 @@
+# dvc.api.make_checkpoint()
+
+Make an
+[in-code checkpoint](/doc/user-guide/experiment-management#checkpoints-in-source-code).
+
+```py
+def make_checkpoint()
+```
+
+#### Usage:
+
+```py
+from dvc.api import make_checkpoint
+
+while True:
+    # ... write a stage output
+    make_checkpoint()
+```
+
+## Description
+
+To track successive steps in a longer
+[experiment](/doc/user-guide/experiment-management), you can write your code so
+it registers checkpoints with DVC during runtime. This function should be called
+by the code in stages executes by `dvc exp run` (see `cmd` field of `dvc.yaml`).
+
+> Note that for non-Python code, the alternative is to write a
+> `.dvc/tmp/DVC_CHECKPOINT` signal file.
+
+## Exceptions
+
+None
+
+## Example: Mark a checkpoint every 100 iterations
+
+Let's consider the following `dvc.yaml` file:
+
+```yaml
+stages:
```

Comment:
Adapted from https://github.com/iterative/dvc/wiki/Experiments#checkpoint-experiments for now. Need a better example...

Reply:
I'll improve the example in a future iteration (maybe once @dberenbaum's #2195 is merged).